assets/blocks/course-actions-block/course-actions/course-actions-edit.js

```diff
@@ -24,15 +34,69 @@ const innerBlocksTemplate = [
 	],
 ];
 
+/**
+ * Hook for selecting a child block.
+ *
+ * @param {string} clientId Block client ID.
+ * @return {Function} Function to select a child block by block name.
+ */
+const useSelectChildBlock = ( clientId ) => {
+	const select = useSelect( 'core/block-editor' );
```

Comment:
If there's not a special reason in this case, I'd suggest selecting the value directly here to avoid any problem with the tree changing and not triggering the change here.

My suggestion would be something like (not tested):

```js
const { toSelect } = useSelect( ( select ) => {
    const childBlocks = select( blockEditorStore ).getBlock( clientId ).innerBlocks;  // Notice that I also changed the store to `blockEditorStore`.
	const toSelect = childBlocks.find(
		( block ) => block.name === blockName
	);

	return { toSelect };
} );
const { selectBlock } = useDispatch( blockEditorStore );
```

Reply:
Thanks for the suggestion! If I'm understanding correctly, I think part of the problem is that `useSelect` and `useDispatch` have to be used outside of the returned callback (because of the rules of hooks), but I don't have access to the `blockName` variable until I am _inside_ the callback.

I've made a change in 256037dca to move more of the logic outside, so I don't use `select` or `dispatch` variables directly inside of the callback. Does that seem better?

Reply:
> I don't have access to the blockName variable until I am inside the callback

Ah, sorry! I missed the part of the `blockName` variable.

> I think part of the problem is that useSelect and useDispatch have to be used outside of the returned callback (because of the rules of hooks)

The problem wouldn't be the rules of hooks because you were calling it correctly before the `useCallback`. The problem would me more about of how `useSelect` works. If using in the previous way, it wouldn't be triggered again when the store changed, which could cause unexpected future problems.

 But it looks great now! Thank you for the update!

Reply:
> The problem would me more about of how useSelect works. If using in the previous way, it wouldn't be triggered again when the store changed, which could cause unexpected future problems.

Ahh because the actual `select` object is the same, right? This way, if the store changes, then the callback given to `useSelect` will be re-run to get the new `childBlocks` value.

Just trying to make sure I understand haha

Reply:
> This way, if the store changes, then the callback given to useSelect will be re-run to get the new childBlocks value.

Exactly! =) I found a reference about this here just before the Parameters: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useselect